### PR TITLE
add support for snapshot export

### PR DESCRIPTION
**Pull request by illi-j**

### CODE REVIEW

File: snapshots.go [hunk 1]
```diff
@@ -137,3 +137,19 @@ func (RevertSnapshot) Response() interface{} {
 func (RevertSnapshot) AsyncResponse() interface{} {
 	return new(BooleanResponse)
 }
+
+// ExportSnapshot exports an instant snapshot of a volume to SOS
```
Comment (falzm) [inline snapshots.go:141]:
```suggestion
// ExportSnapshot exports a volume snapshot.
```

File: snapshots.go [hunk 2]
```diff
@@ -137,3 +137,19 @@ func (RevertSnapshot) Response() interface{} {
...
+type ExportSnapshot struct {
+	ID *UUID `json:"id"`
+	_  bool  `name:"exportSnapshot" description:"Exports an instant snapshot of a volume."`
+}
+
+// ExportSnapshotResponse represents the presigned URL to access the exported snapshot
```
Comment (falzm) [inline snapshots.go:147]: There will be more information returned besides the pre-signed URL at some point (e.g. the file checksum).

```suggestion
// ExportSnapshotResponse represents the response of a snapshot export operation.
```

File: snapshots.go [hunk 3]
```diff
@@ -137,3 +137,19 @@ func (RevertSnapshot) Response() interface{} {
 func (RevertSnapshot) AsyncResponse() interface{} {
 	return new(BooleanResponse)
 }
+
+// ExportSnapshot exports a volume snapshot
+type ExportSnapshot struct {
+	ID *UUID `json:"id"`
+	_  bool  `name:"exportSnapshot" description:"Exports an instant snapshot of a volume."`
```
Comment (pierre-emmanuelJ) [inline snapshots.go:144]:
```suggestion
	_  bool  `name:"exportSnapshot" description:"Exports an instance snapshot of a volume."`
```
instance?

Comment (falzm) [reply snapshots.go:144]: @pierre-emmanuelJ "instant" (referring to the snapshot of a instantaneous capture of a state) is correct, although a bit overkill but it's used elsewhere in other calls in this package so ¯\\\_(ツ)_/¯

Comment (pierre-emmanuelJ) [reply snapshots.go:144]: Okay I got it ;)

**Pull request merged**